base/applications/rapps/misc.cpp

```diff
@@ -376,3 +413,78 @@ SearchPatternMatch(LPCWSTR szHaystack, LPCWSTR szNeedle)
     /* TODO: Improve pattern search beyond a simple case-insensitive substring search. */
     return StrStrIW(szHaystack, szNeedle) != NULL;
 }
+
+BOOL
+DeleteDirectoryTree(LPCWSTR Dir, HWND hwnd)
+{
+    CStringW from(Dir);
+    UINT cch = from.GetLength();
+    from.Append(L"00");
+    LPWSTR p = from.GetBuffer();
+    p[cch] = p[cch + 1] = L'\0'; // Double null-terminate
```

Comment:
It looks like you could get away with this hack by using the two-parameter version of `Append()`, see
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/atl/atlsimpstr.h;h=ce0286602b4e416a71985bc2c11fcda1a4c00642;hb=605fa32d459e0ad299a64f4db86eefb60fa8d10f#l271
```suggestion
    from.Append(L"\0\0", 2); // Double null-terminate
    LPWSTR p = from.GetBuffer();
```
Let me know whether this works.
(I kept the p = from.GetBuffer() because you use it in the shfos structure init below.)

Reply:
> me know whether this works

I don't know if it works now but I'm not comfortable with it. If someone comes along later and implements that FIXME there it will break this hack for sure.

> Could UNICODE_NULL be of beneficial use here?

That is just a fancy way of saying `L'\0'`